l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -212,6 +225,156 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
         $.operators[msg.sender].validatorCount -= uint128(validatorCount);
     }
 
+    /**
+     * @inheritdoc IUniFiAVSManager
+     * @dev Restricted in this context is like `whenNotPaused` modifier from Pausable.sol
+     */
+    function registerValidatorsOptimistically(ValidatorRegistrationParams[] memory paramsArray)
```

Comment:
I'd rename `paramsArray` to just `validators` and use calldata instead of memory